modules/accredited_representative_portal/app/services/filter_service.rb

```diff
@@ -0,0 +1,48 @@
+class FilterService
+  def self.handle_filter(filter_params)
+    filter_params[:status] ||= 'Pending'
+    filter_params[:sortField] ||= 'ar_power_of_attorney_requests.created_at'
+    filter_params[:sortDirection] ||= 'desc'
+    filter_params[:pageNumber] ||= 0
+    filter_params[:pageSize] ||= 10
+
+    result = QueryParamsSchema.call(filter_params)
+    if result.success?
+      normalized_params = result.to_h
+
+      if normalized_params[:status].present?
+        if normalized_params[:status] == "Pending"
+          where_clause = "resolution.id is null"
+          join_clause = ""
+        else
+            where_clause = "resolution.resolving_type = 'AccreditedRepresentativePortal::PowerOfAttorneyRequestDecision' and decision.type = "
+          if normalized_params[:status] == "Accepted"
+            join_clause = "LEFT OUTER JOIN ar_power_of_attorney_request_decisions AS decision ON decision.id = resolution.resolving_id AND resolution.resolving_type = 'AccreditedRepresentativePortal::PowerOfAttorneyRequestDecision'"
+            where_clause += "'PowerOfAttorneyRequestAcceptance'"
+          elsif normalized_params[:status] == "Declined"
+            join_clause = "LEFT OUTER JOIN ar_power_of_attorney_request_decisions AS decision ON decision.id = resolution.resolving_id AND resolution.resolving_type = 'AccreditedRepresentativePortal::PowerOfAttorneyRequestDecision'"
+            where_clause += "'PowerOfAttorneyRequestDeclination'"
+          end
+        end
```

Comment:
let's move all these query fragments into the AR model as scopes, something like this:

```
  scope :with_status, ->(status) {
    case status
    when 'Pending'
      where('resolution.id IS NULL')
    when 'Accepted'
      joins(:resolution)
        .where("resolution.resolving_type = 'AccreditedRepresentativePortal::PowerOfAttorneyRequestAcceptance'")
    when 'Declined'
      joins(:resolution)
        .where("resolution.resolving_type = 'AccreditedRepresentativePortal::PowerOfAttorneyRequestDeclination'")
    else
      all
    end
  }

  scope :sorted_by, ->(field, direction) {
    order(field => (direction == 'asc' ? :asc : :desc))
  }

  scope :paginate, ->(page, page_size) {
    offset(page * page_size).limit(page_size)
  }
```